src/libraries/analytics/analyticsEventTracking.ts

```diff
@@ -0,0 +1,79 @@
+  HUMAN_ID_SUCCESS_NEW_REGISTRATION = 'OB-Username_na_verified',
+  HUMAN_ID_FAILED_VERIFICATION = 'external_API_failed_verification',
+
+  // ONBOARDING INPUT USERNAME SCREEN
+  ONBOARDING_USERNAME_PROFILE_PIC_CLICKED = 'OB-Username_ProfilePic_clicked',
+  ONBOARDING_USERNAME_PROFILE_PIC_CAMERA_SELECT = 'OB-Username_ProfilePicDrawer_clickCamera',
+  ONBOARDING_USERNAME_PROFILE_PIC_LIBRARY_SELECT = 'OB-Username_ProfilePicDrawer_clickLibrary',
+  ONBOARDING_USERNAME_PROFILE_PIC_IMAGE_CHANGED = 'OB-Username_ProfilePic_updated',
+  ONBOARDING_USERNAME_PROFILE_PIC_IMAGE_FAIL_TO_CHANGE = 'OB-Username_ProfilePic_failed',
+  ONBOARDING_USERNAME_PROFILE_PIC_ALERT_SKIP = 'OB-Username_PicReminderAlert_skip',
+  ONBOARDING_USERNAME_PROFILE_PIC_ALERT_ADD_PHOTO = 'OB-Username_PicReminderAlert_addProfilePic',
+
+  // ONBOARDING SELECT LOCATION SCREEN
+  ONBOARDING_LOCATION_FIRST_OPEN_SEARCH = 'OB-Location_Location1_openSearch',
+  ONBOARDING_LOCATION_FIRST_DELETED = 'OB-Location_Location1_deleted',
+  ONBOARDING_LOCATION_SECOND_OPEN_SEARCH = 'OB-Location_Location2_openSearch',
+  ONBOARDING_LOCATION_SECOND_DELETED = 'OB-Location_Location2_deleted',
+  ONBOARDING_LOCATION_FIRST_SELECTED = 'OB-Location_Location1_hasLocationvalue',
+  ONBOARDING_LOCATION_SECOND_SELECTED = 'OB-Location_Location2_hasLocationvalue',
+  ONBOARDING_LOCATION_DRAWER_CLOSED = 'OB-Location_SearchBottomDrawer_closewithoutselection',
+
+  // ONBOARDING TOPICS SCREEN
+  ONBOARDING_TOPICS_TOTAL_FOLLOWING = 'OB-Topics_TopicsNextButton_clicked',
+
+  // ONBOARDING WHO TO FOLLOW SCREEN
+  ONBOARDING_WHO_TO_FOLLOW_TOTAL_FOLLOWING = 'OB-FollowUsers_UsersNextButton_clicked',
+
+  // ONBOARDING REGISTRATION
+  ONBOARDING_REGISTRATION_UPLOAD_IMAGE_SUCCESS = 'OB-FollowUsers_API_UploadImageSuccess',
+  ONBOARDING_REGISTRATION_UPLOAD_IMAGE_FAIL = 'OB-FollowUsers_API_UploadImageFail',
+  ONBOARDING_REGISTRATION_SUCCESS = 'OB-FollowUsers_API_CreateAccountSuccess',
+  ONBOARDING_REGISTRATION_FAILED = 'OB-FollowUsers_API_CreateAccountFail'
+}
+
+const ENABLE_TOAST = true;
+
+const AnalyticsEventTracking = (() => {
+  const client = createClient({
+    writeKey: SEGMENT_WRITE_KEY || '',
+    debug: __DEV__,
+    trackAppLifecycleEvents: true,
+    trackDeepLinks: true
+  });
+
+  return {
+    eventTrack: (event: BetterSocialEventTracking, additionalData?: object) => {
+      if (!event) throw new Error('Event must be defined');
+      if (additionalData && typeof additionalData !== 'object')
+        throw new Error('additionalData must be an object');
+
+      if (!additionalData) {
+        if (ENABLE_TOAST && !!SEGMENT_WRITE_KEY) SimpleToast.show(event);
+        return client.track(event);
+      }
+
+      if (ENABLE_TOAST && !!SEGMENT_WRITE_KEY)
+        SimpleToast.show(`${event} ${JSON.stringify(additionalData || {})}`);
+      return client.track(event, (additionalData || {}) as JsonMap);
+    }
+  };
+})();
+
+Object.freeze(AnalyticsEventTracking);
+
+export default AnalyticsEventTracking;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The implementation of the analytics event tracking module seems to be well done. However, there are a few points that could be improved:

1. The `ENABLE_TOAST` constant is set to `true`, which means toast notifications will be shown every time an event is tracked. This might not be desirable in a production environment as it could disrupt the user experience. Consider making this value configurable based on the environment (development or production).

2. The error handling in the `eventTrack` function could be improved. Currently, if an error occurs (for example, if `event` is not defined or `additionalData` is not an object), an error is thrown but not caught anywhere. This could potentially crash the application. Consider adding a try-catch block around the code that could throw errors and handle them appropriately.

3. The `SEGMENT_WRITE_KEY` is used without any null or undefined check. If it's not provided, the `createClient` function will receive an empty string as the `writeKey`, which could lead to unexpected behavior. It would be better to add a check for `SEGMENT_WRITE_KEY` before initializing the Segment client.

Here are the suggested changes:

```diff
const ENABLE_TOAST = __DEV__;
...
const AnalyticsEventTracking = (() => {
  if (!SEGMENT_WRITE_KEY) {
    console.error('Segment write key is not provided');
    return;
  }

  const client = createClient({
    writeKey: SEGMENT_WRITE_KEY,
    debug: __DEV__,
    trackAppLifecycleEvents: true,
    trackDeepLinks: true
  });
...
  return {
    eventTrack: (event: BetterSocialEventTracking, additionalData?: object) => {
      try {
        if (!event) throw new Error('Event must be defined');
        if (additionalData && typeof additionalData !== 'object')
          throw new Error('additionalData must be an object');

        if (!additionalData) {
          if (ENABLE_TOAST && !!SEGMENT_WRITE_KEY) SimpleToast.show(event);
          return client.track(event);
        }

        if (ENABLE_TOAST && !!SEGMENT_WRITE_KEY)
          SimpleToast.show(`${event} ${JSON.stringify(additionalData || {})}`);
        return client.track(event, (additionalData || {}) as JsonMap);
      } catch (error) {
        console.error(error);
      }
    }
  };
})();
```
This way, the toast notifications will only be shown in the development environment, and errors thrown in the `eventTrack` function will be caught and logged to the console. Also, a check for `SEGMENT_WRITE_KEY` is added before initializing the Segment client.


<!-- This is an auto-generated comment by OSS CodeRabbit -->